### PR TITLE
Update database diagram using MySQL Workshop

### LINKED ISSUE #7: Update database diagram
We have a database diagram here:
https://github.com/freedomofpress/securedrop-docs/blob/main/docs/diagrams/securedrop-database.png

It is generated from:
https://github.com/freedomofpress/securedrop-docs/blob/main/docs/diagrams/securedrop-database.tex

It is outdated (last updated 2016) and misses schema changes (e.g., addition of UUIDs) and tables added since then. We should update it, and commit to keeping it up-to-date as part of every schema change.

### PR DESCRIPTION
**Pull request by eloquence**
## Status

Ready for review

## Description of Changes

Brings the database diagram up to date and includes cardinalities in crow's foot notation.

This diagram was created using MySQL Workshop, which is open source and cross-platform (while designed for MySQL, there does not appear to be a conflict between its capabilities and our relatively modest requirements). To create it, I first dumped the SQLite database and imported it using the "Import > Reverse Engineer MySQL Create Script" option. The main things I manually adjusted after that are:
- using the "indexes" section to enumerate UNIQUE constraints and foreign keys (seemed like the best fit).
- correcting the source<->star relationship to be 1:1 (side note: why this is a separate table is unclear to me)
- changing TINYINT to BOOLEAN  for readability (both MySQL and SQLite store Booleans as {0,1} integers internally)

Resolves #7.

## Alternatives considered

I evaluated [SQLeton](https://github.com/inukshuk/sqleton) and [ERAlchemy](https://pypi.org/project/ERAlchemy/) for fully automatic diagram generation. Neither choice offered sufficient flexibility to allow for correcting cardinalities, choosing notation styles, optimizing placement, or configuring the display of column flags such as NOT NULL or UNIQUE.

MySQL Workbench seemed a good middle ground between partial automation and partial manual effort.

<details>
<summary>
SQLeton generated diagram
</summary>

SQLeton
</details>

<details>
<summary>
ERAlchemy generated diagram
</summary>

ERAlchemy
</details>

## Review
1. Spin up the SecureDrop Docker development environment via `make dev` in your SecureDrop checkout directory
2. Shell into it via `docker exec -it securedrop-dev-0 bash`
3. Run SQLite via `sqlite3 /var/lib/securedrop/db.sqlite`
4. Output the schema via `.schema`

Compare the output with the diagram `securedrop-database.png` from this PR:
1) Are the table descriptions accurate?
2) Are the relationships (including cardinalities) accurate?

## Checklist (Optional)

- [ ] Doc linting (`make docs-lint`) passed locally
- [ ] You have previewed (`make docs`) docs at http://localhost:8000

### CODE REVIEW

File: docs/development/tips_and_tricks.rst [hunk 1]
```diff
@@ -131,7 +131,9 @@ Note that the ``cat`` example above will also add the ATHS info for the
 Architecture Diagrams
 ---------------------
```
Comment (eloquence) [inline docs/development/tips_and_tricks.rst:132]: As a side note, this section is extremely buried in our docs right now, suggestions for reorganizing welcome for a follow-up PR :)

File: docs/development/tips_and_tricks.rst [hunk 2]
```diff
@@ -131,7 +131,9 @@ Note that the ``cat`` example above will also add the ATHS info for the
...
 
-Some helpful diagrams for getting a sense of the SecureDrop application architecture are stored `here
-<https://github.com/freedomofpress/securedrop/tree/develop/docs/diagrams>`_, including a high-level view of the SecureDrop database structure:
+Some helpful diagrams for getting a sense of the SecureDrop application
+architecture are stored `here <https://github.com/freedomofpress/securedrop-docs/tree/main/docs/diagrams>`_,
+including a high-level view of the SecureDrop database structure:
 
 .. image:: ../diagrams/securedrop-database.png
+  :width: 100%
```
Comment (eloquence) [inline docs/development/tips_and_tricks.rst:139]: This ensures that clicking the diagram will display the full-size version.

**Comment by creviera** [comment]
Seems like Primary isn't showing what the primary key is, right?

primary-key

**Comment by creviera** [comment]
Oh, I see, the little yellow light bulb is the indicator. Just weird to see "PRIMARY" by itself there.

**Comment by eloquence** [comment]
I think it's supposed to be a key icon. Yeah, mysql-workbench doesn't let you name that in the "indexes" section, presumably because "there can be only one" (cue Highlander theme).

**Comment by creviera** [comment]
> correcting the source<->star relationship to be 1:1 (side note: why this is a separate table is unclear to me)

**Comment by creviera** [comment]
^ oops i keep pressing Ctrl+Enter because SecureDrop client is training me to!! 

> correcting the source<->star relationship to be 1:1 (side note: why this is a separate table is unclear to me)
I don't think it normalizes anything and could be moved into the sources table

**Review by creviera — APPROVED** [review]
lgtm!

**Pull request merged**